pkg/tecdsa/gg20/participant/rounds_test_bench_test.go

```diff
@@ -0,0 +1,414 @@
+			if i != j {
+				cosigners = append(cosigners, j)
+			}
+		}
+
+		err := signers[i].setCosigners(cosigners)
+		require.NoError(t, err)
+
+		if useDistributed {
+			r1P2pIn = make(map[uint32]*Round1P2PSend, playerMin)
+
+			for j := uint32(1); j <= uint32(playerMin); j++ {
+				if i != j {
+					r1P2pIn[j] = r1P2P[j][i]
+				}
+			}
+		}
+
+		params := map[uint32]*Round1Bcast{}
+
+		for j := uint32(1); j <= uint32(playerMin); j++ {
+			if i != j {
+				params[j] = signerOut[j]
+			}
+		}
+
+		p2p[i], err = signers[i].SignRound2(params, r1P2pIn)
+		require.NoError(t, err)
+	}
+
+	return p2p
+}
+
+func signRound3(t *testing.T, signers map[uint32]*Signer, p2p map[uint32]map[uint32]*P2PSend, playerMin int) map[uint32]*Round3Bcast {
+	var err error
+
+	round3Bcast := make(map[uint32]*Round3Bcast, playerMin)
+
+	for i := uint32(1); i <= uint32(playerMin); i++ {
+		signP2P := map[uint32]*P2PSend{}
+
+		for j := uint32(1); j <= uint32(playerMin); j++ {
+			if i != j {
+				signP2P[j] = p2p[j][i]
+			}
+		}
+
+		round3Bcast[i], err = signers[i].SignRound3(signP2P)
+		require.NoError(t, err)
+	}
+
+	return round3Bcast
+}
+
+func signRound4(t *testing.T, signers map[uint32]*Signer, round3Bcast map[uint32]*Round3Bcast, playerMin int) map[uint32]*Round4Bcast {
+	var err error
+
+	round4Bcast := make(map[uint32]*Round4Bcast, playerMin)
+
+	for i := uint32(1); i <= uint32(playerMin); i++ {
+		signP2P := map[uint32]*Round3Bcast{}
+
+		for j := uint32(1); j <= uint32(playerMin); j++ {
+			if i != j {
+				signP2P[j] = round3Bcast[j]
+			}
+		}
+
+		round4Bcast[i], err = signers[i].SignRound4(signP2P)
+		require.NoError(t, err)
+	}
+
+	return round4Bcast
+}
+
+func signRound5(t *testing.T, curve elliptic.Curve, pk *curves.EcPoint, signers map[uint32]*Signer, round4Bcast map[uint32]*Round4Bcast, playerMin int) (map[uint32]*Round5Bcast, map[uint32]map[uint32]*Round5P2PSend) {
+	var (
+		err error
+
+		round5Bcast = make(map[uint32]*Round5Bcast, playerMin)
+		r5P2p       = make(map[uint32]map[uint32]*Round5P2PSend, playerMin)
+	)
+
+	for i := uint32(1); i <= uint32(playerMin); i++ {
+		witnesses := map[uint32]*Round4Bcast{}
+
+		for j := uint32(1); j <= uint32(playerMin); j++ {
+			if i != j {
+				witnesses[j] = round4Bcast[j]
+			}
+		}
+
+		round5Bcast[i], r5P2p[i], err = signers[i].SignRound5(witnesses)
+		require.NoError(t, err)
+	}
+
+	rBark := signers[1].state.Rbark
+
+	for i := uint32(2); i <= uint32(playerMin); i++ {
+		rBark, err = rBark.Add(signers[i].state.Rbark)
+		require.NoError(t, err)
+	}
+
+	rBark.Y, err = core.Neg(rBark.Y, curve.Params().P)
+	require.NoError(t, err)
+
+	rBark, err = rBark.Add(pk)
+	require.NoError(t, err)
+
+	if !rBark.IsIdentity() {
+		t.Errorf("%v != %v", rBark.X, pk.X)
+		t.FailNow()
+	}
+
+	return round5Bcast, r5P2p
+}
+
+func signRound6(t *testing.T, msg []byte, signers map[uint32]*Signer, round5Bcast map[uint32]*Round5Bcast, r5P2P map[uint32]map[uint32]*Round5P2PSend, playerMin int, useDistributed bool) {
```

Comment:
So it's both offline + online 6 round?

Reply:
Yes

![image](https://user-images.githubusercontent.com/16573297/159909979-f158a737-fcd4-401f-b30a-feeb670a60da.png)
![image](https://user-images.githubusercontent.com/16573297/159910003-8f066319-eef6-41de-8e32-9c927642364f.png)
![image](https://user-images.githubusercontent.com/16573297/159910034-cf662410-821c-4001-b8e0-59f25f7bf7b6.png)

In this method msg is used two times at the beginning, i think that only the last `SignOutput` call could be considered as an offline round

Reply:
![image](https://user-images.githubusercontent.com/16573297/159910365-bd51b37b-9f90-4f01-a0ae-3e4eaa580b28.png)

Found that splitting only inside the `SignRound6Full` method

Reply:
This splitting is only kryptology's optimization, other implementations make last round fully online, so online round measurements with only `signRound6Online` should be faster (at least we can hope so)

Reply:
Added metrics for only `signRound6Online`